test/gurka/test_match.cc

```diff
@@ -44,4 +44,39 @@ TEST(Standalone, BasicMatch) {
   EXPECT_NEAR(len, 100.f, 1.f);
 
   gurka::assert::raw::expect_path_length(result, 0.100, 0.001);
+}
+
+TEST(Standalone, UturnMatch) {
+  midgard::logging::GetLogger();
+
+  const std::string ascii_map = "A--1--2--B";
+
+  const gurka::ways ways = {
+      {"AB", {{"highway", "primary"}}},
+  };
+
+  const auto layout = gurka::detail::map_to_coordinates(ascii_map, 10);
+  auto map = gurka::buildtiles(layout, ways, {}, {}, "test/data/uturn_match");
+
+  auto result =
+      gurka::match(map, {"1", "2", "1", "2"}, false, "auto", R"({"penalize_immediate_uturn":false})");
```

Comment:
start in the middle of the edge, go a little bit, turn around in the middle of the edge, go a bit, turn around in the middle of the edge, end